core/execution/engine.go

```diff
@@ -1013,12 +1011,12 @@ func (e *Engine) OnTick(ctx context.Context, t time.Time) {
 	toDelete := []string{}
 	parentStates := e.getParentStates()
 	evts := make([]events.Event, 0, len(e.marketsCpy))
-	rejected := map[int]struct{}{}
+	maxI := len(e.marketsCpy) - 1
 	for i, mkt := range e.marketsCpy {
-		if _, ok := rejected[i]; ok {
-			// successor markets were rejected, because of how golang iterates over slices, this loop
-			// will still iterate over rejected markets, these markets must be skipped
-			continue
+		// when a successor market kicks in, the marketsCpy slice is updated
```

Comment:
Can't we just collect the to-reject and reject them after we've finished the loop? Seems generally better than fiddling with the slice while iterating over it.

Reply:
Thought of that, but it'd end up being more verbose (we'd have to check every market we iterate over, to see if we should handle the time update or not), and then add an additional loop over that map to reject the markets in question. This way avoids a second loop, results in the same behaviour, and allows for market succession to be handled all in a single block (including cleaning up the checkpoint state etc...).

Reply:
More verbose but less risky.

Also won't we already be on-ticking children that should be rejected if they happen to appear before the child that comes out of auction first in the marketsCpy?

Reply:
If they appear before the successor, then the `OnTick` is fine. They may be ready to leave opening auction, but in that case, it's the oldest successor that takes priority, so older pending successors must be checked first.